deps/sharedlibpp/src/SharedLibraryFactory.cpp

```diff
@@ -64,6 +84,38 @@ shlibpp::SharedLibraryFactory::Private::Private(int32_t startCheck,
     memset(&api, 0, sizeof(SharedLibraryClassApi));
 }
 
+std::vector<std::string> shlibpp::SharedLibraryFactory::Private::platformSpecificLibName(const std::string& library)
+{
+
+#if defined(_WIN32)
+    return {library + ".dll", library + "d.dll", "lib" + library + ".dll"};
+#elif defined(__linux__)
+    return {"lib" + library + ".so"};
+#elif defined(__APPLE__)
+    return {"lib" + library + ".dylib"};
+#endif
```

Comment:
I would add a clear last `elif` with a compilation error in case the library is compiled in a non supported OS.

Reply:
:+1: